docs/patterns/components/Table/index.js

```diff
@@ -19,63 +19,110 @@ const tableComponentNames = [
   'TableRow'
 ];
 
-const TableDemo = () => (
-  <Demo
-    component={Table}
-    customImport={`import {\n  ${tableComponentNames.join(
-      ',\n  '
-    )}\n} from '@deque/cauldron-react'`}
-    states={[
-      {
-        children: (
-          <>
-            <TableHead>
-              <TableRow>
-                <TableHeader scope="col">First Name</TableHeader>
-                <TableHeader scope="col">Last Name</TableHeader>
-                <TableHeader scope="col">Email</TableHeader>
-              </TableRow>
-            </TableHead>
-            <TableBody>
-              <TableRow>
-                <TableCell>Frank</TableCell>
-                <TableCell>Zappa</TableCell>
-                <TableCell>frank@zappa.io</TableCell>
-              </TableRow>
-              <TableRow>
-                <TableCell>Duane</TableCell>
-                <TableCell>Allman</TableCell>
-                <TableCell>duane@almond.biz</TableCell>
-              </TableRow>
-              <TableRow>
-                <TableCell>Yamandu</TableCell>
-                <TableCell>Costa</TableCell>
-                <TableCell>yamandu_costa@gmail.br</TableCell>
-              </TableRow>
-              <TableRow>
-                <TableCell>Jimmy</TableCell>
-                <TableCell>Herring</TableCell>
-                <TableCell>jamesHerring@hotmail.gov</TableCell>
-              </TableRow>
-            </TableBody>
-          </>
-        )
-      }
-    ]}
-    propDocs={{
-      children: {
-        ...children,
-        required: true
-      },
-      className
-    }}
-  >
-    <p>
-      <strong>NOTE:</strong> the table width by default is set to 100%. If a
-      max-width is desired, you can simply use your own CSS to apply it.
-    </p>
-  </Demo>
-);
+const sampleData = [
+  {
+    first_name: 'Frank',
+    last_name: 'Zappa',
+    email: 'frank@zappa.io'
+  },
+  {
+    first_name: 'Duane',
+    last_name: 'Allman',
+    email: 'duane@almond.biz'
+  },
+  {
+    first_name: 'Yamandu',
+    last_name: 'Costa',
+    email: 'yamandu_costa@gmail.br'
+  },
+  {
+    first_name: 'Jimmy',
+    last_name: 'Herring',
+    email: 'jamesHerring@hotmail.gov'
+  }
+];
+
+const TableDemo = () => {
+  const [sort, setSort] = useState([null, 'none']);
+  const [sortBy, sortDir] = sort;
+
+  const comparator = data =>
+    data.sort((a, b) =>
+      (sortDir === 'ascending' ? a[sortBy] : b[sortBy]).localeCompare(
+        sortDir === 'ascending' ? b[sortBy] : a[sortBy]
+      )
+    );
+
+  const sortedData = sortBy ? comparator(sampleData) : sampleData;
+
+  const getSortDir = column => (column === sortBy ? sortDir : 'none');
+  /* asc => desc
+     none => asc
+     desc => asc */
+  const setSortDir = column =>
+    column === sortBy && sortDir === 'ascending' ? 'descending' : 'ascending';
```

Comment:
This function name is a little misleading, since it doesn't actually set the sort direction. I think it would be less confusing (and more useful) if this function actually did set the sort direction. It could then be called with `setSortDir('last_name')` instead of doing `setSort(['last_name', setSortDir('last_name')])`